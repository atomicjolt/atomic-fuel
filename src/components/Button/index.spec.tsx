import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button, ButtonType } from '.';

describe('Button component', () => {
  const renderComponent = (onClick = () => {}) => {
    const ariaOptions = {
      'aria-label': 'survey options',
      'aria-haspopup': 'menu',
      'aria-expanded': false,
    };

    return render(
      <Button
        ariaOptions={ariaOptions}
        color="gray"
        buttonType={ButtonType.icon}
        onClick={onClick}
      >
        Click
      </Button>,
    );
  };
  it('Should match default snapshot', () => {


    const { asFragment } = renderComponent();

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should click the button', () => {
    const onClick = jest.fn();
    renderComponent(onClick);
    fireEvent.click(screen.getByText('Click'));
    expect(onClick).toHaveBeenCalled();
  });
});
