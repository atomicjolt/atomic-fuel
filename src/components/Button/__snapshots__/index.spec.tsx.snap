// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should match default snapshot 1`] = `
<DocumentFragment>
  <button
    aria-expanded="false"
    aria-haspopup="menu"
    aria-label="survey options"
    class="aj-btn aj-btn--gray aj-btn--icon"
    type="button"
  >
    Click
  </button>
</DocumentFragment>
`;
